File: 0048-rotate-image/0048-rotate-image.go
func rotate(matrix [][]int)  {
    xLength := len(matrix)
    yLength := len(matrix[0])
    
    for i := 0; i < xLength; i++ {
        for j := i; j < yLength; j++ {
            temp := matrix[i][j]
            matrix[i][j] = matrix[j][i]
            matrix[j][i] = temp
        }
    }
    
    for i := 0; i < xLength; i++ {
        for j := 0; j < yLength / 2; j++ {
            temp := matrix[i][j]
            matrix[i][j] = matrix[i][yLength - 1 -j]
            matrix[i][yLength - 1 - j] = temp
        }
    }
}

File: 0048-rotate-image/0048-rotate-image.java
class Solution {
    public void rotate(int[][] matrix) {
        int xLength = matrix.length;
        int yLength = matrix[0].length;
        
        for (int i = 0; i < xLength; i++) {
            for (int j = i; j < yLength; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        
        for (int i = 0; i < xLength; i++) {
            for (int j = 0; j < yLength / 2; j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[i][yLength - j - 1];
                matrix[i][yLength - j - 1] = temp;
            }
        }
    }
}